### PR TITLE
Add Local Storage to Contact Form

### PR DESCRIPTION
**Pull request by saadat-ali-awan**
### Project requirements
- You should implement the following interactions:
  - When the user changes the content of any input field, the data is saved to the local storage.
  - When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- You should use the following data model:
  - You must create **a single JavaScript object with all the data from the entire form** and save it in local storage. Do not create one local storage entry per input field.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+    if (document.querySelector('small')) form.removeChild(document.querySelector('small'));
+    if (form.email.value !== form.email.value.toLowerCase()) {
+      event.preventDefault();
+      const listItem = document.createElement('li');
+      const errorMessage = document.createElement('small');
+      errorMessage.innerText = 'Email Contains Capital Letters';
+      listItem.append(errorMessage);
+      form.querySelector('ul').append(listItem);
+    }
+  });
```
Comment (andersonlebon) [inline js/main.js:44]: ## [OPTIONAL]
- [ ] Kindly I would suggest you use to fix this issue on your validation. Currently when I double click on the submit button this is being submitted although your email input contains an uppercase letter.

File: js/main.js [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
+
+  document.querySelectorAll('input, textarea').forEach((inp) => {
+    inp.addEventListener('change', () => {
+      const inputName = inp.getAttribute('name');
+      formData[inputName] = inp.value;
+      saveData(formData);
+    });
+  });
 });
```
Comment (andersonlebon) [inline js/main.js:53]: ## [OPTIONAL]
- [ ] This is not in the requirements is very important. KIndly I would suggest you add your past milestones to this project. Currently, your popup functionality is not working. Kindly make sure you add it to this milestone.

image

**Review by andersonlebon — APPROVED** [review]
## Hi @Saadat123456 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by sheylaPozo** [comment]
## Status: ✔️ Project Approved ✔️ 
----

Hello @Saadat123456 🔥 

> # Excellent work on your project, it looks very professional and it's completed. Congrats on the good work!

surprised

Now you can celebrate! 🥇 🚀 👍 💯 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️
`There's nothing left to say but time to merge! 😊`

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**